modules/host/update.go

```diff
@@ -58,27 +104,182 @@ func (h *Host) ProcessConsensusChange(cc modules.ConsensusChange) {
-	}
-	for _, block := range cc.AppliedBlocks {
-		// TODO: Look for transactions relevant to open storage obligations.
-
-		// Adjust the height of the host. The host height is initialized to
-		// zero, but the genesis block is actually height zero. For the genesis
-		// block only, the height will be left at zero.
-		//
-		// Checking the height here eliminates the need to initialize the host
-		// to and underflowed types.BlockHeight.
-		if block.ID() != types.GenesisBlock.ID() {
-			h.blockHeight++
+		for _, block := range cc.AppliedBlocks {
+			// Look for transactions relevant to open storage obligations.
+			for _, txn := range block.Transactions {
+				// Check for file contracts.
+				if len(txn.FileContracts) > 0 {
+					for i := range txn.FileContracts {
+						fcid := txn.FileContractID(uint64(i))
+						so, err := getStorageObligation(tx, fcid)
+						if err != nil {
+							// The storage folder may not exist, or the disk
+							// may be having trouble. Either way, we ignore the
+							// problem. If the disk is having trouble, the user
+							// will have to perform a rescan.
+							continue
+						}
+						so.OriginConfirmed = true
+						err = putStorageObligation(tx, so)
+						if err != nil {
+							continue
+						}
+					}
+				}
+
+				// Check for file contract revisions.
+				if len(txn.FileContractRevisions) > 0 {
+					for _, fcr := range txn.FileContractRevisions {
+						so, err := getStorageObligation(tx, fcr.ParentID)
+						if err != nil {
+							// The storage folder may not exist, or the disk
+							// may be having trouble. Either way, we ignore the
+							// problem. If the disk is having trouble, the user
+							// will have to perform a rescan.
+							continue
+						}
+						so.RevisionConfirmed = true
+						err = putStorageObligation(tx, so)
+						if err != nil {
+							continue
+						}
+					}
+				}
+
+				// Check for storage proofs.
+				if len(txn.StorageProofs) > 0 {
+					for _, sp := range txn.StorageProofs {
+						so, err := getStorageObligation(tx, sp.ParentID)
+						if err != nil {
+							// The storage folder may not exist, or the disk
+							// may be having trouble. Either way, we ignore the
+							// problem. If the disk is having trouble, the user
+							// will have to perform a rescan.
+							continue
+						}
+						so.ProofConfirmed = true
+						err = putStorageObligation(tx, so)
+						if err != nil {
+							continue
+						}
+					}
+				}
+			}
+
+			// Height is not adjusted when dealing with the genesis block because
+			// the default height is 0 and the genesis block height is 0. If adding
+			// the genesis block, height will already be at height 0 and should not
+			// update.
+			if block.ID() != types.GenesisBlock.ID() {
+				h.blockHeight++
+			}
+
+			// Handle any action items relevant to the current height.
+			bai := tx.Bucket(bucketActionItems)
+			heightBytes := make([]byte, 8)
+			binary.BigEndian.PutUint64(heightBytes, uint64(h.blockHeight))
```

Comment:
I think there is, but doesn't hurt to have the same information in multiple places.